src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -5149,15 +5157,15 @@ void func_8083AE40(Player* this, s16 objectId) {
 void func_8083AF44(PlayState* play, Player* this, s32 magicSpell) {
     func_80835DE4(play, this, func_808507F4, 0);
 
-    this->unk_84F = magicSpell - 3;
+    this->unk_84F = magicSpell - PLAYER_MAGIC_SPELL(PLAYER_IA_FARORES_WIND);
     Magic_RequestChange(play, sMagicSpellCosts[magicSpell], MAGIC_CONSUME_WAIT_PREVIEW);
 
     LinkAnimation_PlayOnceSetSpeed(play, &this->skelAnime, &gPlayerAnim_link_magic_tame, 0.83f);
 
-    if (magicSpell == 5) {
+    if (magicSpell == PLAYER_MAGIC_SPELL(PLAYER_IA_DINS_FIRE)) {
         this->subCamId = OnePointCutscene_Init(play, 1100, -101, NULL, CAM_ID_MAIN);
     } else {
-        func_80835EA4(play, 10);
+        func_80835EA4(play, KEEPON4_ITEM_TYPE_10);
```

Comment:
```suggestion
        func_80835EA4(play, CAM_ITEM_TYPE_10);
```
I feel since this is external to camera, it should have a `CAM_` prefix to it. Maybe add onto it the setting it's associated with, `TURN_AROUND`, but I'm not sure it's necessary. I don't think `CAM_ITEM_TYPE_` conflicts with anything, although that alone might not be clear enough.

Also, now that you understand this function much more, do you think `CAM_SET_TURN_AROUND` is decent, or is there something better than `TURN_AROUND` that could be used for this? (And possibly be incorporated into enums/function names i.e. `func_80835EA4`)

Reply:
Yeah TURN_AROUND isn't the best name tbh (neither is item_type, just got that from the debug string as placeholder, not to mention "keepon4")

I may want to refer to this whole thing (turn_around, keepon4, item_type) as "player action camera" but I'm not sure that's not too generic of a name for it, it may encompass more things I'm not aware of

Reply:
Agreed about "TURN_AROUND", never felt quite right to me. 

I feel the list I had is pretty comprehensive of all use-cases of this camera settings:
• 0x1: Reading ruto’s letter, dropping fish
• 0x2: Drinking from a bottle
• 0x3: Releasing fairy
• 0x4: Show empty item, catch item in bottle, FW Return
• 0x8: Grabbing ruto’s letter from underwater (or any generic GI from surfacing from the water)
• 0x9: Receive item from npc, grabbing freestanding item, get item from chest, get most items
• 0x5: Releasing bugs and blue fire
• 0xA: Casting FW and Nayru’s Love
• 0xC: Link's Death cutscene
• 0x5A: Ocarina without target
• 0x5B: Ocarina with target (Great Fairy Fountain, botw, frogs) (Not zora river, not skull kid, not oca game)
• 0x51: Immediately after playing epona,
• 0xB: Navi talking to you

Reply:
The commonality between them is that the camera turns around and faces Link from the front instead of being behind Link

Reply:
I'm really tempted by "focus player action" but that overlaps with onepointdemo stuff (not that there is a name that doesn't though...)

I get that interpretation of "turn around", but it's more refering to the data KeepOn4 typically uses than what it can actually do
(i.e. if it used angles that made it not face Link's front, it wouldn't)

Maybe if this stays open for 6 more months I'll have an idea by then but I don't plan to find good names for `CAM_SET_TURN_AROUND`/`CAM_ITEM_TYPE`/`KeepOn4` in this PR

Reply:
I was thinking about this, and maybe instead of "turn around", what about "inverted camera"? I know "inverted cam" is the trick used to back-walk with the camera facing Link, maybe that term could be expanded to KeepOn4 and the associated player functions/enums. Thoughts?

Reply:
Im not too sure, the glitch we call "inverted camera" is called as such cause the camera is "inverted" from how its supposed to be while targeting or in "parralel" mode. Im not sure it carries over well to a case where the camera is intended to be facing player.

I tbh find "turn around" to be quite intuitive. Its not only the fact that it faces link, but also includes the little swinging or "turning around" motion it does when getting there.
But if we want to move away from it, I would probably suggest going down the path of "front of player" "face player front", or something along those lines